app.js

```diff
@@ -189,51 +191,91 @@ class Player {
         // always set the last dose timer to the current time so that players do not accumulate dosage while the game is closed
         this.lastDoseTimestamp = Date.now();
         this.currentDamageLevel = saved?.player?.currentDamageThreshold ?? 0;
-        this.symptoms = saved?.player?.symptoms ?? new Set([]);
+        this.symptoms = new Set(saved?.player?.symptoms ?? []);
 
         // Knowledge is a set of known words.
         // We don't (yet) keep a full translation table, in either direction;
         // we just obfuscate unknown words.
         // We also keep "\n" here so we can preserve newlines in the input >.>
-        this.knowledge =
-            saved?.player?.knowledge ?? new Set([]);
+        this.knowledge = new Set(saved?.player?.knowledge ?? []);
+    }
+
+    save() {
+        return {
+            location: this.location,
+            dosage: this.dosage,
+            damage: this.damage,
+            currentDamageLevel: this.currentDamageLevel,
+            symptoms: Array.from(this.symptoms),
+            knowledge: Array.from(this.knowledge),
+        }
+    }
+}
+
+class Item {
+    constructor(itemid, permanent, saved) {
+        const perm = permanent.items[itemid];
+        const save = (saved ? saved[itemid] : {})
+        // Immutable properties:
+        this.aliases = perm.aliases
+        this.moveable = perm.moveable
+        this.passive = perm.passive
+        this.sense = perm.sense
+        this.action = perm.action
+        this.writing = perm.writing
+        this.lightLevel = perm.lightLevel;
+
+        // Saved properties:
+        this.rosetta = save?.rosetta ?? perm.rosetta;
```

Comment:
I didn't instantiate `writtenWords` on an item before, but now with save state we should. Could also refactor `rosetta` back to immutable and combine `rosetta` and `writtenWords` for translation?